File: wonseok970/Programmers/행렬 테두리 회전하기.py
from collections import deque


def solution(rows, columns, queries):
    arr = [[0] * columns for _ in range(rows)]
    cnt = 0
    answer = []
    for i in range(rows):
        for j in range(columns):
            cnt += 1
            arr[i][j] = cnt

    for qur in queries:
        q = deque([])
        x1, y1, x2, y2 = qur

        for i in range(y1 - 1, y2):
            q.append(arr[x1 - 1][i])

        for i in range(x1, x2):
            q.append(arr[i][y2 - 1])

        for i in range(y2 - 2, y1 - 2, -1):
            q.append(arr[x2 - 1][i])

        for i in range(x2 - 2, x1 - 1, -1):
            q.append(arr[i][y1 - 1])

        q.rotate(1)
        answer.append(min(q))

        for i in range(y1 - 1, y2):
            arr[x1 - 1][i] = q.popleft()

        for i in range(x1, x2):
            arr[i][y2 - 1] = q.popleft()

        for i in range(y2 - 2, y1 - 2, -1):
            arr[x2 - 1][i] = q.popleft()

        for i in range(x2 - 2, x1 - 1, -1):
            arr[i][y1 - 1] = q.popleft()

    return answer

File: gettls/week24/행렬 테두리 회전하기.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int arr[101][101];

vector<int> solution(int rows, int columns, vector<vector<int>> queries) {
    vector<int> answer;
    map<int, int> m;
    int n = 1;
    for (int i = 1; i <= rows; i++) {
        for (int j = 1; j <= columns; j++) {
            arr[i][j] = n++;
        }
    }

    for (int i = 0; i < queries.size(); i++) {
        int x1 = queries[i][0], y1 = queries[i][1];
        int x2 = queries[i][2], y2 = queries[i][3];
        int x = x1, y = y1;
        int start = arr[x][y], min = start;

        while (x < x2) {
            if (arr[x + 1][y] < min) min = arr[x + 1][y];
            arr[x][y] = arr[x + 1][y];
            x++;
        }
        while (y < y2) {
            if (arr[x][y + 1] < min) min = arr[x][y + 1];
            arr[x][y] = arr[x][y + 1];
            y++;
        }
        while (x > x1) {
            if (arr[x - 1][y] < min) min = arr[x - 1][y];
            arr[x][y] = arr[x - 1][y];
            x--;
        }
        while (y > y1) {
            if (arr[x][y - 1] < min) min = arr[x][y - 1];
            arr[x][y] = arr[x][y - 1];
            y--;
        }
        arr[x][y + 1] = start;
        answer.push_back(min);
    }

    return answer;
}